packages/cli/src/util/dev/server.ts

```diff
@@ -1579,7 +1579,9 @@ export default class DevServer {
               const devServerParsed = new URL(this.address);
               if (devServerParsed.origin === rewriteUrlParsed.origin) {
                 // remove origin, leaving the path
-                req.url = rewritePath.slice(rewriteUrlParsed.origin.length);
+                req.url =
+                  rewritePath.slice(rewriteUrlParsed.origin.length) || '/';
+                prevUrl = req.url;
```

Comment:
I think this doesn't matter because the `prevUrl` would stay a full url while `req.url` would be a relative one, but for complete correctness (which may prevent future bugs), I think we should still update `prevUrl` here.